`make install-tools-local` fails without GitHub CLI installed and authenticated

`make install-tools-local` fails without GitHub CLI installed and authenticated.

The requirement to have the GitHub CLI (gh) installed and authenticated to set up the Argo CD build environment is not documented in the Developer's Guide.

<!-- If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack). -->

Checklist:

- [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
- [X] I've included steps to reproduce the bug.
- [ ] I've pasted the output of `argocd version`.

**To Reproduce**

1. Run without the `gh` tool installed:
```
$ which gh
$ make install-tools-local
...
. hack/tool-versions.sh && gh extension install --pin v${GH_AW_VERSION} --force github/gh-aw
/bin/sh: 1: gh: not found
make: *** [Makefile:648: install-gh-aw-local] Error 127
```

2. Run with a gh instance that is not authenticated:

```
$ which gh
/usr/bin/gh
$ gh auth status
You are not logged into any GitHub hosts. To log in, run: gh auth login
$ make install-tools-local
...
. hack/tool-versions.sh && gh extension install --pin v${GH_AW_VERSION} --force github/gh-aw
To get started with GitHub CLI, please run:  gh auth login
Alternatively, populate the GH_TOKEN environment variable with a GitHub API authentication token.
make: *** [Makefile:648: install-gh-aw-local] Error 4
```

<!-- A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue. -->

Expected behavior

The command should not fail, or at least the dependency on the GitHub CLI should be documented in the Developer's Guide.

IMO, having authenticated gh in the developer's environment should be optional. It is not required to build, test, run, or debug Argo CD proper, but only to work on GitHub Agentic Workflows, which, I assume, is mostly of interest to Argo CD maintainers rather than a regular Argo CD user or contributor.

**Version**

```shell
Argo CD from `master` branch
```

--- a/Makefile
+++ b/Makefile
@@ -286,12 +286,15 @@ resourceiconsgen:
 	hack/generate-icons-typescript.sh
 
 .PHONY: codegen-local
-codegen-local: mod-vendor-local gogen protogen clientgen openapigen clidocsgen mockgen actionsdocsgen resourceiconsgen manifests-local notification-docs notification-catalog gh-aw-compile
+codegen-local: mod-vendor-local gogen protogen clientgen openapigen clidocsgen mockgen actionsdocsgen resourceiconsgen manifests-local notification-docs notification-catalog
 	rm -rf vendor/
 
 .PHONY: codegen-local-fast
-codegen-local-fast: gogen protogen-fast clientgen openapigen clidocsgen mockgen manifests-local notification-docs notification-catalog gh-aw-compile
+codegen-local-fast: gogen protogen-fast clientgen openapigen clidocsgen mockgen manifests-local notification-docs notification-catalog
 
+# Recompiles the GitHub Agentics workflow lock files after editing issue-triage.md or aw.json.
+# Only needed by maintainers working on GitHub Agentics workflows; requires an installed and
+# authenticated gh CLI plus the gh-aw extension (see install-gh-aw-local).
 .PHONY: gh-aw-compile
 gh-aw-compile:
 	gh aw compile
@@ -640,11 +643,17 @@ show-go-version: test-tools-image
 
 # Installs all tools required to build and test ArgoCD locally
 .PHONY: install-tools-local
-install-tools-local: install-test-tools-local install-codegen-tools-local install-go-tools-local install-gh-aw-local
+install-tools-local: install-test-tools-local install-codegen-tools-local install-go-tools-local
 
-# Installs the gh aw CLI extension for managing GitHub Agentics workflows
+# Installs the gh aw CLI extension for managing GitHub Agentics workflows.
+# Only needed by maintainers working on GitHub Agentics workflows; not part of install-tools-local.
+# Requires an installed and authenticated gh CLI.
 .PHONY: install-gh-aw-local
 install-gh-aw-local:
+	@if ! command -v gh >/dev/null 2>&1; then \
+		echo "gh CLI is required to install the gh-aw extension. See https://cli.github.com/ and https://argo-cd.readthedocs.io/en/latest/developer-guide/github-agentics-issue-triage/"; \
+		exit 1; \
+	fi
 	. hack/tool-versions.sh && gh extension install --pin v$${GH_AW_VERSION} --force github/gh-aw
 
 # Installs all tools required for running unit & end-to-end tests (Linux packages)
@@ -760,3 +769,7 @@ help:
 	@echo '  codegen(-local) -- if using -local, run the following targets first'
 	@echo '  install-codegen-tools-local -- run this to install the codegen tools'
 	@echo '  install-go-tools-local -- run this to install go libraries for codegen'
+	@echo
+	@echo 'github agentics (maintainers only):'
+	@echo '  install-gh-aw-local -- install the gh aw CLI extension (requires an authenticated gh CLI)'
+	@echo '  gh-aw-compile -- recompile the agentics workflow lock files after editing issue-triage.md or aw.json'

--- a/docs/developer-guide/github-agentics-issue-triage.md
+++ b/docs/developer-guide/github-agentics-issue-triage.md
@@ -12,12 +12,18 @@ Github Agentics is essentially a GitHub workflow, which is created and updated b
 
 The initial configuration was performed by running `gh aw add-wizard githubnext/agentics/issue-triage` locally, this was a one-time step. 
 
+## Prerequisites
+
+> [!NOTE]
+> These tools are only needed by maintainers working on GitHub Agentics workflows. They are **not** required to build, test, run, or debug Argo CD itself, and are therefore **not** installed by `make install-tools-local`.
+
+- An installed and authenticated [GitHub CLI (`gh`)](https://cli.github.com/). Authenticate it with `gh auth login` (or by setting the `GH_TOKEN` environment variable).
+- The `gh aw` CLI extension, installed by running `make install-gh-aw-local`.
+
 ## Configuration
 
-A dedicated GitHub `gh aw` CLI needs to be installed locally, by running `make install-gh-aw-local`. The CLI is also installed as part of `make install-tools-local`. 
-
-- `.github/workflows/issue-triage.md` - the main file with which the maintainers can interact to configure the workflow. It contains both a prompt for the agent and a configuration section. The file is pre-created upon the initial installation of the wizard and then it can be configured further. Upon performing changes in this file, it is required to run `gh aw compile` or `make codegen-local`. 
-- `.github/workflows/aw.json` - additional configuration file. Upon performing changes in this file, it is required to run `gh aw compile` or `make codegen-local`.
+- `.github/workflows/issue-triage.md` - the main file with which the maintainers can interact to configure the workflow. It contains both a prompt for the agent and a configuration section. The file is pre-created upon the initial installation of the wizard and then it can be configured further. Upon performing changes in this file, it is required to run `gh aw compile` or `make gh-aw-compile`. 
+- `.github/workflows/aw.json` - additional configuration file. Upon performing changes in this file, it is required to run `gh aw compile` or `make gh-aw-compile`.
 
 ## Auto generated workflow files
 - `.github/workflows/issue-triage.lock.yml` - the GitHub CI workflow itself, auto-generated based on the `issue-triage.md` and `aw.json` files. 

